pkg/autodiscovery/integration/config.go

```diff
@@ -264,13 +264,25 @@ func (c *Config) Digest() string {
 		inst := RawMap{}
 		err := yaml.Unmarshal(i, &inst)
 		if err != nil {
+			log.Debugf("Error while calculating config digest, skipping: %v", err)
 			continue
 		}
-		tagList, _ := inst["tags"].([]string)
-		sort.Strings(tagList)
-		inst["tags"] = tagList
+		if val, found := inst["tags"]; found {
```

Comment:
I think it's worth adding a comment here to explain that the purpose of all this logic is to sort the list of `tags` (so that the digest is stable for 2 configs that have the same tags in a different order).

I think it's also worth adding a test that checks this specifically (i.e. 2 configs with the same tags ordered differently produces same digest)